spec/models/users_model_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+describe User, type: :model do
+  describe 'validation tests' do
+    let!(:user1){ create :user, email: 'dupa@dupa.pl' }
+    it 'ensure user is valid' do
+      expect(user1).to be_valid
+    end
+    context 'when name of user is too long' do
+      it 'rails validation error' do
+        user1.name = "a" * 51
+        expect(user1).to_not be_valid
+        expect(user1.errors.count).to eq 1
+        expect(user1.errors[:name]).to eq ["is too long (maximum is 50 characters)"]
+      end
+    end
+    context 'when name of user is not presence' do
+      it 'rails validation error' do
+        user1.name = nil
+        expect(user1).to_not be_valid
+        expect(user1.errors.count).to eq 1
+        expect(user1.errors[:name]).to eq ["can't be blank"]
+      end
+    end
+    context 'when email address is not unique' do
+      let!(:user2) do
+        described_class.new(name: "Piecia", email: "Dupa@dupa.pl", password: "123",
+                            phone: "123456789")
+      end
+      it 'rails validation error' do
+        expect(user2).to_not be_valid
+        expect(user2.errors[:email]).to eq ["has already been taken"]
+        expect(user2.errors.count).to eq 1
+      end
+    end
+    context 'when email address is too long' do
+      it 'rails validation error' do
+        user1.email = 'a' * 256 + '@darek.pl'
+        expect(user1).to_not be_valid
+        expect(user1.errors[:email]).to eq ["is too long (maximum is 255 characters)"]
+        expect(user1.errors.count).to eq 1
+      end
+    end
+    context 'when email address is invalid' do
```

Comment:
Podobnie jak wyżej, niepotrzebny podział na contexty, gdzie context jest w sumie ten sam (ten sam setup danych, dwa różne testy na tych danych). Albo podziel na ity jak wskazałem wyżej, albo rób niepoprawny/pusty email poza testem, tylko w setupie danych jak to robisz dwa testy wyżej w "when email address is not unique"
Fajniej by było, jakby były konsystentnie testy napisane, jeśli dotyczą tego samego.
Plus niżej też jest rails co jest dziwne trochę, raczej wiemy na czym pracujemy, nie trzeba tego w teście opisywać.